pom.xml

```diff
@@ -220,7 +219,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.14.1</version>
+                <version>${compiler.plugin.version}</version>
                 <configuration>
                     <release>17</release>
                 </configuration>
```

Comment:
Yo may also want to add the maven enforcer plugin to check the compiler for being at least Java 17.